lib/rails_admin_import/formats/file_importer.rb

```diff
@@ -0,0 +1,37 @@
+module RailsAdminImport
+  module Formats
+    class FileImporter
+      def initialize(import_model, params)
+        if params.has_key?(:file)
+          @filename = params[:file].tempfile
+        end
+        @import_model = import_model
+      end
+
+      attr_reader :filename, :error, :import_model
+
+      def valid?
+        if filename.nil?
+          @error = I18n.t("admin.import.missing_file")
+          false
+        else
+          true
+        end
+      end
+
+      def each(&block)
+        return enum_for(:each) unless block_given?
+
+        if RailsAdminImport.config.logging && filename
+          FileUtils.copy(filename, File.join(Rails.root, "log", "import", "#{Time.now.strftime("%Y-%m-%d-%H-%M-%S")}-import.csv"))
```

Comment:
Line is too long. [130/80]<br>Prefer single-quoted strings inside interpolations.